framework/src/modules/interoperability/mainchain/module.ts

```diff
@@ -431,4 +450,102 @@ export class MainchainInteroperabilityModule extends BaseInteroperabilityModule
 			throw new Error('Invalid validatorsHash from chainData.lastCertificate.');
 		}
 	}
+
+	// https://github.com/LiskHQ/lips/blob/main/proposals/lip-0045.md#mainchain
+	private _verifyTerminatedStateAccounts(
+		chainInfos: ChainInfo[],
+		terminatedStateAccounts: TerminatedStateAccountWithChainID[],
+	) {
+		for (const chainInfo of chainInfos) {
+			const { chainID } = chainInfo;
+
+			if (terminatedStateAccounts.filter(a => a.chainID.equals(chainID)).length > 0) {
```

Comment:
This check is ok. (checking that if for all chainID with a terminated account, the status is "TERMINATED"). But we should add also a check that there can not be a terminated account if there is no chain account (see the [if and only if](https://github.com/LiskHQ/lips/blob/6796693a0aaf97c8a7dd63cbc73f923ca5a39f53/proposals/lip-0045.md?plain=1#L1563) in the LIP).

Concretely, I think that this would mean here that if `terminatedStateAccounts.filter(a => a.chainID.equals(chainID)).length === 0`, then an error should be thrown.